casper/src/main/scala/coop/rchain/casper/util/comm/CasperPacketHandler.scala

```diff
@@ -30,127 +30,164 @@ import monix.execution.Scheduler
-        // or at least the simplified 0 required sigs genesis ceremony.
-        if (conf.approveGenesis) {
-          for {
-            walletsFile <- Genesis
-                            .toFile[F](conf.walletsFile, conf.genesisPath.resolve("wallets.txt"))
-            wallets   <- Genesis.getWallets[F](walletsFile, conf.walletsFile)
-            timestamp <- conf.deployTimestamp.fold(Time[F].currentMillis)(_.pure[F])
-            bondsFile <- Genesis.toFile[F](conf.bondsFile, conf.genesisPath.resolve("bonds.txt"))
-            bonds <- Genesis
-                      .getBonds[F](bondsFile, conf.numValidators, conf.genesisPath)
-            validatorId <- ValidatorIdentity.fromConfig[F](conf)
-            bap = new BlockApproverProtocol(
-              validatorId.get,
-              timestamp,
-              bonds,
-              wallets,
-              conf.minimumBond,
-              conf.maximumBond,
-              conf.hasFaucet,
-              conf.requiredSigs
-            )
-            gv <- Ref.of[F, CasperPacketHandlerInternal[F]](
-                   new GenesisValidatorHandler(
-                     runtimeManager,
-                     validatorId.get,
-                     conf.shardId,
-                     bap
-                   )
-                 )
-          } yield new CasperPacketHandlerImpl[F](gv)
-        } else if (conf.createGenesis) {
-          for {
-            genesis <- Genesis.fromInputFiles[F](
-                        conf.bondsFile,
-                        conf.numValidators,
-                        conf.genesisPath,
-                        conf.walletsFile,
-                        conf.minimumBond,
-                        conf.maximumBond,
-                        conf.hasFaucet,
-                        runtimeManager,
-                        conf.shardId,
-                        conf.deployTimestamp
+        val msg    = "ApprovedBlock not found in storage, connecting to existing network"
+        val action = connectAndQueryApprovedBlock[F](init)
+        (msg, action)
+    } >>= {
+      case (msg, action) => Log[F].info(msg) >> action
+    }
+
+  def connectToExistingNetwork[F[_]: LastApprovedBlock: Metrics: BlockStore: ConnectionsCell: NodeDiscovery: TransportLayer: ErrorHandler: RPConfAsk: SafetyOracle: Sync: Concurrent: Time: Log: MultiParentCasperRef: BlockDagStorage](
+      approvedBlock: ApprovedBlock,
+      init: CasperInit[F]
+  ): F[CasperPacketHandler[F]] =
+    for {
+      validatorId <- ValidatorIdentity.fromConfig[F](init.conf)
+      genesis     = approvedBlock.candidate.flatMap(_.block).get
+      casper <- MultiParentCasper.hashSetCasper[F](
+                 init.runtimeManager,
+                 validatorId,
+                 genesis,
+                 init.conf.shardId
+               )
+      _                   <- MultiParentCasperRef[F].set(casper)
+      _                   <- Log[F].info("Making a transition to ApprovedBlockRecievedHandler state.")
```

Comment:
It's a good moment to fix the `Recieved` typo

Reply:
hopefully no integration tests rely on it :D